### ISSUE TITLE
💅 `noInvalidUseBeforeDeclaration` - false-positive for param destructuring with default values

### ISSUE THREAD

**Issue by ipanasenko** [biomejs/biome#1648]
### Environment information

```bash
CLI:
  Version:                      1.5.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.18.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/9.8.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

noInvalidUseBeforeDeclaration

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAYQBGAHUAbgBjAHQAaQBvAG4AIAA9ACAAKAB7AGEAIAA9ACAAJwAxADEAMQAnACwAIABiACAAPQAgAGEAfQApACAAPQA%2BACAAewAKACAAIABjAG8AbgBzAG8AbABlAC4AaQBuAGYAbwAoAGEALABiACkAOwAKAH0ACgBjAG8AbgBzAG8AbABlAC4AaQBuAGYAbwAoAGEARgB1AG4AYwB0AGkAbwBuACkAOwA%3D

### Expected result

No linting error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ipanasenko** [comment]
Related to https://github.com/biomejs/biome/issues/1652
